cmd/servicelog/post.go

```diff
@@ -24,175 +26,208 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var (
+type PostCmdOptions struct {
 	Message         servicelog.Message
 	ClustersFile    servicelog.ClustersFile
-	template        string
+	Template        string
 	filterFiles     []string // Path to filter file
 	filtersFromFile string   // Contents of filterFiles
 	isDryRun        bool
 	skipPrompts     bool
 	clustersFile    string
 	internalOnly    bool
+	Args            []string
 
 	// Messaged clusters
-	successfulClusters = make(map[string]string)
-	failedClusters     = make(map[string]string)
-)
+	successfulClusters map[string]string
+	failedClusters     map[string]string
+}
 
-const (
-	defaultTemplate = ""
-)
+func newPostCmd() *cobra.Command {
+	var opts = PostCmdOptions{
+		successfulClusters: make(map[string]string),
+		failedClusters:     make(map[string]string),
+	}
+	postCmd := &cobra.Command{
+		Use:   "post CLUSTER_ID",
+		Short: "Send a servicelog message to a given cluster",
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			opts.Args = args
+			cmdutil.CheckErr(opts.Init())
+			cmdutil.CheckErr(opts.Validate())
+			cmdutil.CheckErr(opts.Run())
+		},
+	}
+
+	// define required flags
+	postCmd.Flags().StringVarP(&opts.Template, "template", "t", "", "Message template file or URL")
+	postCmd.Flags().StringArrayVarP(&templateParams, "param", "p", templateParams, "Specify a key-value pair (eg. -p FOO=BAR) to set/override a parameter value in the template.")
```

Comment:
How would a user send a servicelog that required parameters to be specified?

Reply:
Good catch on this.

There's a "templateParams" global variable in `/cmd/servicelog/common.go` that doesn't really need to be common. I can move that into the "post" command's struct and make it an extracted property so callers can set it.